

drivers/input/keyboard/atkbd.c: In function `atkbd_do_set_extra':
drivers/input/keyboard/atkbd.c:225: warning: passing arg 4 of `atkbd_attr_set_helper' from incompatible pointer type
drivers/input/keyboard/atkbd.c: In function `atkbd_do_set_scroll':
drivers/input/keyboard/atkbd.c:226: warning: passing arg 4 of `atkbd_attr_set_helper' from incompatible pointer type
drivers/input/keyboard/atkbd.c: In function `atkbd_do_set_set':
drivers/input/keyboard/atkbd.c:227: warning: passing arg 4 of `atkbd_attr_set_helper' from incompatible pointer type
drivers/input/keyboard/atkbd.c: In function `atkbd_do_set_softrepeat':
drivers/input/keyboard/atkbd.c:228: warning: passing arg 4 of `atkbd_attr_set_helper' from incompatible pointer type
drivers/input/keyboard/atkbd.c: In function `atkbd_do_set_softraw':
drivers/input/keyboard/atkbd.c:229: warning: passing arg 4 of `atkbd_attr_set_helper' from incompatible pointer type


Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/drivers/input/keyboard/atkbd.c |   10 ++++++----
 1 files changed, 6 insertions(+), 4 deletions(-)

diff -puN drivers/input/keyboard/atkbd.c~atkbd-warning-fixes drivers/input/keyboard/atkbd.c
--- 25/drivers/input/keyboard/atkbd.c~atkbd-warning-fixes	2004-10-01 22:25:30.265809264 -0700
+++ 25-akpm/drivers/input/keyboard/atkbd.c	2004-10-01 22:25:30.269808656 -0700
@@ -206,8 +206,9 @@ struct atkbd {
 
 static ssize_t atkbd_attr_show_helper(struct device *dev, char *buf,
 				ssize_t (*handler)(struct atkbd *, char *));
-static ssize_t atkbd_attr_set_helper(struct device *dev, const char *buf, size_t count,
-				int (*handler)(struct atkbd *, const char *, size_t));
+static ssize_t
+atkbd_attr_set_helper(struct device *dev, const char *buf, size_t count,
+		ssize_t (*handler)(struct atkbd *, const char *, size_t));
 #define ATKBD_DEFINE_ATTR(_name)						\
 static ssize_t atkbd_show_##_name(struct atkbd *, char *);			\
 static ssize_t atkbd_set_##_name(struct atkbd *, const char *, size_t);		\
@@ -925,8 +926,9 @@ out:
 	return retval;
 }
 
-static ssize_t atkbd_attr_set_helper(struct device *dev, const char *buf, size_t count,
-				int (*handler)(struct atkbd *, const char *, size_t))
+static ssize_t
+atkbd_attr_set_helper(struct device *dev, const char *buf, size_t count,
+		ssize_t (*handler)(struct atkbd *, const char *, size_t))
 {
 	struct serio *serio = to_serio_port(dev);
 	struct atkbd *atkbd;
_
